tests/cypress/e2e/hedy_page/save_share_button.cy.js

```diff
@@ -0,0 +1,59 @@
+import {goToHedyPage, goToHome} from "../tools/navigation/nav";
+import {loginForStudent} from "../tools/login/login";
+
+describe('Checks if save & share button works', () => {
+    it('Passes', () => {
+      loginForStudent();
+      cy.get('#start_programming_button').click();
+
+      // type in editor 'print test'
+      cy.get('#editor').type('\nprint test');
+      
+      cy.get('#runit').click();
+      cy.wait(3000);
+
+      // the save and share button:
+      cy.get('#share_program_button')
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click();
+      
+      // this is for if you already have saved your level once:
+      if (cy.get('#modal-confirm')){
+        cy.get('#modal-yes-button').click();
+      }
+
+      // block for copying link should be visible:
+      cy.get('#modal-copy')
+      .should('be.visible');
+
+      // clicking on button for copying the share link
+      cy.get('#modal-copy-button')
+      .should('be.visible')
+      .click();
+
+      // checking if correct link is copied to clipboard:
+      //http://localhost:8080/hedy/d89964c395f647bfb07c0ef4bcce8f31/view (this is the correct link)
+      cy.window().then((win) => {
+        win.navigator.clipboard.readText().then((text) => {
+          expect(text).include('/hedy/d89964c395f647bfb07c0ef4bcce8f31/view');
```

Comment:
I think (correct me if I'm wrong) that the program string is created dynamically one save/share and is different each time. You can find the code that creates this code at `website/programs.py#105` where the program id is set by getting a random hex: `program_id = uuid.uuid4().hex`.

Reply:
this cypress test is a javascript file, but the programs is a python file. Is it possible to import a python variable into a javascript file?

Reply:
I think that would be quite cumbersome. The easiest solution to completely verify this flow would be:
- Save/share the program
- Navigate to `/programs`
- Get the copy link of the last program (as we just saved it it is the last one)
- Logout
- Navigate to the link, verify that it works

Reply:
okay noted! I will implement this.

Reply:
I cannot seem to fetch the clipboard content (which contains the link) in order to navigate to this link. I will try to fix the other comments, but I don't think I can resolve this copied link part before our project deadline (which is Monday).

Reply:
It is stored as an `onclick` attribute on the `modal-copy-button`. You can find the code at `app.ts#637`:

```
if (shared) {
      $('#modal-copy-button').attr('onclick', "hedyApp.copy_to_clipboard('" + viewProgramLink(response.id) + "', '" + response.share_message + "')");
```